src/model/validation/message.ts

```diff
@@ -39,6 +41,30 @@ export class MessageLocation {
             new SourcePosition(loc.end, loc.endToken.line, loc.endToken.column + loc.endToken.end - loc.endToken.start));
     }
 
+    get start(): SourcePosition {
+        if(this._start instanceof  SourcePosition) {
+            return this._start;
+        }
+        if(!this.source){
+            throw new Error("Can not calculate start position without reference to source");
```

Comment:
would be good to check this in the constructor, too

Reply:
@crowgames this one is for you. Do MessageLocations without source === undefined make sense?

Reply:
MessageLocations without `source` (but with a `sourceName`) could be created by the `check-graphql-syntax` validator when no ProjectSource was found matching the GraphQL source. But in that case, the location will be passed to the constructor.